File: 1695-maximum-erasure-value/1695-maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        summ = 0
        left = 0
        st = set()
        n = len(nums)
        maxi = 0
        for right in range(n):
            while nums[right] in st:
                st.remove(nums[left])
                summ -= nums[left]
                left += 1
            st.add(nums[right])
            summ += nums[right]
            maxi = max(maxi, summ)
        return maxi

File: 1695-maximum-erasure-value/1695-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        set<int> st;
        int n = nums.size();
        int left = 0;
        int maxi = 0;
        int sum = 0;
        for(int right=0; right<n; right++){

            while(st.count(nums[right])){
                st.erase(nums[left]);
                sum -= nums[left];
                left++;
            }

            st.insert(nums[right]);
            sum += nums[right];
            maxi = max(maxi, sum);
        }
        return maxi;
    }
};